mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/massindexing/MassIndexingDefaultCleanOperation.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.mapper.pojo.massindexing;
+
+import java.lang.invoke.MethodHandles;
+
+import org.hibernate.search.engine.cfg.spi.ParseUtils;
+import org.hibernate.search.mapper.pojo.logging.impl.Log;
+import org.hibernate.search.util.common.logging.impl.LoggerFactory;
+
+public enum MassIndexingDefaultCleanOperation {
+
+	/**
+	 * Removes all entities from the indexes before indexing.
+	 *
+	 * @see org.hibernate.search.mapper.pojo.massindexing.spi.PojoMassIndexer#purgeAllOnStart(boolean)
+	 */
+	PURGE( "purge" ) {
+		@Override
+		public boolean purge() {
+			return true;
+		}
+
+		@Override
+		public boolean dropAndCreate() {
+			return false;
+		}
+	},
+	/**
+	 * Drops the indexes and their schema (if they exist) and re-creates them before indexing.
+	 *
+	 * @see org.hibernate.search.mapper.pojo.massindexing.spi.PojoMassIndexer#dropAndCreateSchemaOnStart(boolean)
+	 */
+	DROP_AND_CREATE( "drop_and_create" ) {
+		@Override
+		public boolean purge() {
+			return false;
+		}
+
+		@Override
+		public boolean dropAndCreate() {
+			return true;
+		}
+	};
+
+	private static final Log log = LoggerFactory.make( Log.class, MethodHandles.lookup() );
+
+	// This method conforms to the MicroProfile Config specification. Do not change its signature.
+	public static MassIndexingDefaultCleanOperation of(String value) {
+		return ParseUtils.parseDiscreteValues(
+				MassIndexingDefaultCleanOperation.values(),
+				MassIndexingDefaultCleanOperation::externalRepresentation,
+				log::invalidMassIndexingDefaultCleanOperation,
+				value
+		);
+	}
+
+	private final String externalRepresentation;
+
+	MassIndexingDefaultCleanOperation(String externalRepresentation) {
+		this.externalRepresentation = externalRepresentation;
+	}
+
+	/**
+	 * @return The expected string representation in configuration properties.
+	 */
+	public String externalRepresentation() {
+		return externalRepresentation;
+	}
+
+	public abstract boolean purge();
+
+	public abstract boolean dropAndCreate();
```

Comment:
`Enum#equals` works fine, you know :)

Reply:
😮 what? no waaay 😄 

yeah I didn't like these long lines 😃 but at the same time I probably went a bit too far with this 😄 
```java
purgeAtStart = MassIndexingDefaultCleanOperationoperation.PURGE.equals(operation);
dropAndCreateSchemaOnStart = MassIndexingDefaultCleanOperationoperation.DROP_AND_CREATE.equals(operation);
```
will make the change, thanks!